chain-impl-mockchain/src/leadership/genesis/vrfeval.rs

```diff
@@ -18,14 +22,40 @@ impl Nonce {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum FError {
+    InvaildValue(Milli),
+}
+
+impl Display for FError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            FError::InvaildValue(v) => write!(f, "Invalid value {}, should be in range (0,1]", v),
+        }
+    }
+}
+
+impl Error for FError {}
+
 /// number between 0.0 and 1.0 that is used to calculate to
-pub struct F(f64);
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct F(Milli);
```

Comment:
The type documentation could be better and actually refer to the paper that describes F.